apps/studio/components/interfaces/Realtime/Inspector/RealtimeFilterPopover/TableSelector.tsx

```diff
@@ -40,114 +41,141 @@ const TableSelector = ({
+        <PopoverTrigger_Shadcn_ asChild>
+          <Button
+            size={size}
+            type="outline"
+            className={`w-full [&>span]:w-full ${size === 'small' ? 'py-1.5' : ''}`}
+            iconRight={
+              isLoading ? (
+                <IconLoader className="animate-spin" size={12} />
+              ) : (
                 <IconCode className="text-foreground-light rotate-90" strokeWidth={2} size={12} />
-              }
-            >
+              )
+            }
+          >
+            {initiallyLoaded ? (
               <div className="w-full flex space-x-3">
                 <p className="text-xs text-light">table</p>
                 <p className="text-xs">
                   {selectedTableName === '*' ? 'All tables' : selectedTableName}
                 </p>
               </div>
-            </Button>
-          </PopoverTrigger_Shadcn_>
-          <PopoverContent_Shadcn_ className="p-0 w-64" side="bottom" align="start">
-            <Command_Shadcn_>
-              <CommandInput_Shadcn_ placeholder="Find table..." />
-              <CommandList_Shadcn_>
-                <CommandEmpty_Shadcn_>No tables found</CommandEmpty_Shadcn_>
-                <CommandGroup_Shadcn_>
-                  <ScrollArea className={(entities || []).length > 7 ? 'h-[210px]' : ''}>
-                    <CommandItem_Shadcn_
-                      key="all-tables"
-                      className="cursor-pointer flex items-center justify-between space-x-2 w-full"
-                      onSelect={() => {
-                        onSelectTable('*', undefined)
-                        setOpen(false)
-                      }}
-                      onClick={() => {
-                        onSelectTable('*', undefined)
-                        setOpen(false)
-                      }}
-                    >
-                      <span>All tables</span>
-                      {selectedSchemaName === '*' && (
-                        <IconCheck className="text-brand" strokeWidth={2} />
+            ) : (
+              <p className="flex text-xs text-light">Loading tables...</p>
+            )}
+          </Button>
+        </PopoverTrigger_Shadcn_>
+        <PopoverContent_Shadcn_ className="p-0 w-64" side="bottom" align="start">
+          <Command_Shadcn_>
+            <CommandInput_Shadcn_
+              placeholder="Find table..."
+              onValueChange={(str) => searchTables(str)}
+            />
+            <CommandList_Shadcn_>
+              {isLoading && (
+                <div className="flex items-center justify-center space-x-2 px-3 py-2">
+                  <IconLoader className="animate-spin" size={12} />
+                  <p className="flex text-xs text-light">Loading tables...</p>
+                </div>
+              )}
+
+              {showError && isError && (
+                <Alert_Shadcn_ variant="warning" className="!px-3 !py-3">
```

Comment:
i think we can remove the alert's border radius and border - looks weird to have nested borders 

![image](https://github.com/supabase/supabase/assets/19742402/399cf18d-fc15-4c84-9c1b-63aec6f24dba)